main.cpp

```diff
@@ -275,296 +300,283 @@ struct Bakery
     float cutoffFreq = 13200.2f;
-    //2) resonance (float)
     float resonance = 0.77f;
-    //3) filter type (std::string)
     std::string filterType = "Low Pass";
-    //4) input gain (double)
     double inputGain = -5.3;
-    //5) overdrive (float)
     float overdriveAmount = 1.7f;
 
-    //1) cut low frequencies
     void cutLows(float cutoffFreq, float resonance);
-    //2) cut high frequencies
     void cutHighs(float cutoffFreq, float resonance);
-    //3) self-resonate
-    bool selfResonate(float resonance, double inputGain);
+    bool selfResonate(float fResonance, double fInputGain);
 };
 
-/*
-Thing 7) envelope
-5 properties:
-    1) attack time (float)
-    2) decay time (float)
-    3) sustain level (double)
-    4) release time (float)
-    5) curve (float)
-3 things it can do:
-    1) create long sustained signals
-    2) create short staccato signals
-    3) loop over time
- */
+void Filter::cutLows(float fCutoffFreq, float fResonance)
+{
+    std::cout << "Cutting low frequencies below: " << fCutoffFreq << "with a Q of: " << fResonance << std::endl;
+}
+void Filter::cutHighs(float fCutoffFreq, float fResonance)
+{
+    std::cout << "Cutting high frequencies above: " << fCutoffFreq << "with a Q of: " << fResonance << std::endl;
+}
+bool Filter::selfResonate(float fResonance, double fInputGain)
+{
+    if (fResonance > 0.9f && fInputGain > -35.0)
+        return true;
+    else
+        return false;
+}
 
 struct Envelope
 {
-    //1) attack time (float)
     float attackTime = 22.5f;
-    //2) decay time (float)
     float decayTime = 330.2f;
-    //3) sustain level (double)
     double sustainLevel = 0.99228;
-    //4) release time (float)
     float releaseTime = 675.1f;
-    //5) curve (float)
     float curve = 0.9f;
 
-    //1) create long sustained signals
-    void padPreset(float curve);
-    //2) create short staccato signals
-    void staccatoPreset(float curve);
-    //3) loop over time
-    bool loopOver(float attackTime, float decayTime, float releaseTime);
+    void padPreset(float eCurve);
+    void staccatoPreset(float eCurve);
+    bool loopOver(float eAttackTime, float eDecayTime, float eReleaseTime);
 };
 
-/*
-Thing 8) lfo
-5 properties:
-    1) fade-in time (float)
-    2) frequency (double)
-    3) shape (int)
-    4) retrigger behaviour (int)
-    5) mono / poly switch (bool)
-3 things it can do:
-    1) reset
-    2) modulate
-    3) change shape
- */
+void Envelope::padPreset(float eCurve)
+{
+    std::cout << "Setting envelope to slow setting, with curve of: " << eCurve << std::endl;
+}
+void Envelope::staccatoPreset(float eCurve)
+{
+    std::cout << "Setting envelope to fast setting, with curve of: " << eCurve << std::endl;
+}
+bool Envelope::loopOver(float eAttackTime, float eDecayTime, float eReleaseTime)
+{
+    bool eSwitch = false;
+
+    if ((eAttackTime+eDecayTime+eReleaseTime) < 700.f)
+    {
+        eSwitch = true;
+        return true;
+    }
+    else
+        return false;
+}
```

Comment:
add spaces around your math operators so the code is more readable
revise the `if/else return value` as explained early.
```
- [ ] revise